core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -269,22 +270,23 @@ protected ArtifactSignerProvider createArtifactSignerProvider(
                                 baseConfig.getKeyStoreConfigFileMaxSize())));
             registerSignerLoadingHealthCheck(KEYS_CHECK_CONFIG_FILE_LOADING, results);
             signers.addAll(results.getValues());
-          }
 
-          if (azureKeyVaultParameters.isAzureKeyVaultEnabled()) {
-            LOG.info("Bulk loading keys from Azure key vault ... ");
-            /*
-             Note: Azure supports 25K bytes per secret. https://learn.microsoft.com/en-us/azure/key-vault/secrets/about-secrets
-             Each raw bls private key in hex format is approximately 100 bytes. We should store about 200 or fewer
-             `\n` delimited keys per secret.
-            */
-            final MappedResults<ArtifactSigner> azureResult = loadAzureSigners();
-            LOG.info(
-                "Keys loaded from Azure: [{}], with error count: [{}]",
-                azureResult.getValues().size(),
-                azureResult.getErrorCount());
-            registerSignerLoadingHealthCheck(KEYS_CHECK_AZURE_BULK_LOADING, azureResult);
-            signers.addAll(azureResult.getValues());
+            if (azureKeyVaultParameters.isAzureKeyVaultEnabled()) {
+              LOG.info("Bulk loading keys from Azure key vault ... ");
+              /*
+               Note: Azure supports 25K bytes per secret. https://learn.microsoft.com/en-us/azure/key-vault/secrets/about-secrets
+               Each raw bls private key in hex format is approximately 100 bytes. We should store about 200 or fewer
+               `\n` delimited keys per secret.
+              */
+              final MappedResults<ArtifactSigner> azureResult =
+                  loadAzureSigners(azureKeyVaultFactory);
+              LOG.info(
+                  "Keys loaded from Azure: [{}], with error count: [{}]",
+                  azureResult.getValues().size(),
+                  azureResult.getErrorCount());
+              registerSignerLoadingHealthCheck(KEYS_CHECK_AZURE_BULK_LOADING, azureResult);
+              signers.addAll(azureResult.getValues());
+            }
```

Comment:
Just checking this brace is moving the azure bulk loading into the try-with-resources so that it can autoclose AzureVaultFactory?

Does it make sense for the code below this to go in the try too?

Reply:
Yeah, that's exactly why it's moved up so that it can access the AzureVaultFactory that now autocloses using the try-with. The other code below doesn't need to do any closing so I left it alone. It would hurt though move it into the same try-with that made the code easier to read